app/models/line_item.rb

```diff
@@ -3,6 +3,10 @@ class LineItem < ApplicationRecord
   belongs_to :product
   belongs_to :cart, optional: true
 
+  validates :product, uniqueness: { scope: :cart, message: "One product can be added only once in the cart" }, 
```

Comment:
Do we need `cart.present?` check here???

Reply:
Need to discuss this